src/pip/_internal/cli/cmdoptions.py

```diff
@@ -539,6 +544,15 @@ def only_binary() -> Option:
     dest="platforms",
     metavar="platform",
     action="append",
+    choices=[
+        "any",
+        "manylinux1_x86_64",
+        "manylinux1_i386",
+        "manylinux2014_aarch64",
+        "win_amd64",
+        "macosx_10_9_x86_64",
+        "macosx_11_0_arm64",
+    ],
```

Comment:
There are too many platforms to list and we can not restrict users to those listed here.

Reply:
Right. When user input other word, it shouldn't throw any error.